tests/app/main/views/test_suppliers.py

```diff
@@ -10,12 +10,175 @@
 from freezegun import freeze_time
 from lxml import html
 
-from dmtestutils.api_model_stubs import FrameworkStub, SupplierStub
+from dmtestutils.api_model_stubs import (
+    FrameworkStub,
+    SupplierStub,
+    SupplierFrameworkStub,
+)
 from dmtestutils.fixtures import valid_pdf_bytes
 
 from ...helpers import LoggedInApplicationTest, Response
 
 
+class TestSupplierDetailsView(LoggedInApplicationTest):
+
+    def setup_method(self, method):
+        super().setup_method(method)
+        self.user_role = "admin-ccs-data-controller"
+        self.data_api_client_patch = mock.patch('app.main.views.suppliers.data_api_client', autospec=True)
+        self.data_api_client = self.data_api_client_patch.start()
+
+    def teardown_method(self, method):
+        self.data_api_client_patch.stop()
+        super().teardown_method(method)
+
+    def test_get_returns_404_if_supplier_id_is_invalid(self):
+        assert self.client.get("/admin/suppliers/invalid").status_code == 404
+        assert self.client.get("/admin/suppliers/-1").status_code == 404
+        self.data_api_client.get_supplier_frameworks.side_effect = HTTPError(Response(404))
```

Comment:
I don't really understand what these two lines (38/39) are testing. As far as I can see, you're testing that it would work, except you're altering the mock to return a 404. But then doesn't that just test that `mock` returns a 404 when you ask it to?

Reply:
Hmm, it's more demonstrative than a test; the scenario is that the supplied supplier id does not exist. In this case get_supplier_frameworks will return a 404 error. As you say it is a bit redundant, but I think it's important to check that part of the contract; we want to make sure the view doesn't do something strange like turn the 404 into a 503.

Reply:
Let's add in an assert for the api client call 👍

Reply:
Will do

Reply:
Done